### PR TITLE
feat:protect form

### LINKED ISSUE #92: create protect form
- landing page and protect route have 2 separate "protect form" however the functionality is the same.

create one form and make use of it in both page.

### PR DESCRIPTION
**Pull request by huseyincansoylu**
Close #92

### CODE REVIEW

File: packages/nextjs/components/Footer.tsx [hunk 1]
```diff
@@ -25,12 +25,12 @@ export const Footer = () => {
         flexDirection: "row",
         flexGrow: 0,
         bottom: 0,
-        backgroundColor: "#1a5e73", // turquoise[100] + slate[50]
+        backgroundColor: "#ffffff", // turquoise[100] + slate[50]
```
Comment (juanmanso) [inline packages/nextjs/components/Footer.tsx:28]: since the comment no longer holds (because we changed the color), could we remove it please to avoid the noice?

File: packages/nextjs/components/Header/index.tsx [hunk 2]
```diff
@@ -28,13 +28,13 @@ const NavLink = ({ children, href }: PropsWithChildren<{ href: string }>) => {
 export const Header = () => {
   const navLinks = (
     <Row as="ul" style={{ width: undefined }}>
-      <li>
-        <NavLink href="/protect">Protect</NavLink>
+      <li style={{ padding: "0px 5px 0px 0px" }}>
```
Comment (juanmanso) [inline packages/nextjs/components/Header/index.tsx:31]: question: a gap wouldn't suffice here? padding seems odd but maybe you have a good reason for it that I'm not seeing 😅

File: packages/nextjs/components/LargeLogo.tsx [hunk 3]
```diff
@@ -11,7 +11,7 @@ export const LargeLogo = ({ isLight = false }) => {
     <Row as="button" className="gap-2" onClick={browseToHome}>
       <BotBlockIcon height={16} width={16} color={isLight ? coreColors.white : coreColors.black} />
       <Text color={isLight ? "light" : "dark"} type="h3">
-        BotBlock | by Keyko powered by NVM
+        BotBlock | Built by Keyko powered by Nevermined
```
Comment (juanmanso) [inline packages/nextjs/components/LargeLogo.tsx:14]: nitpick: this will change. You can talk later with @paulrmason about it but just letting you know that it will (we already talked about it, didn't change it because specification was not there)

File: packages/nextjs/pages/index.tsx [hunk 4]
```diff
@@ -20,17 +20,18 @@ const Title = () => {
...
+          Make AI Crawlers pay for your content...
+        </Text>
+        <Text type="subheading" style={{ marginBottom: "10px", textAlign: "center" }}>
+          Protect your site from AI crawlers from getting your content for their datasets.
         </Text>
         <Text type="subheading" style={{ textAlign: "center" }}>
-          With BotBlock you can protect your site from AI crawlers from getting your content for their datasets. When
-          your site is protected, you can create a subscription plan for AI companies to pay for access in just a couple
-          of clicks!
+          Once protected, create a subscription plan for AI companies to pay for access with just a couple of clicks!
```
Comment (juanmanso) [inline packages/nextjs/pages/index.tsx:38]: same here, copy was validated before and will probably change so would not bother if I were you

File: packages/nextjs/pages/index.tsx [hunk 5]
```diff
@@ -117,12 +118,12 @@ const SubscriptionOverviewSection = () => {
...
   return (
-    <div style={{ display: "flex", flexDirection: "column", padding: "48px" }}>
-      <div className="flex flex-row justify-start items-center">
+    <div style={{ display: "flex", flexDirection: "column", padding: "20px 0px" }}>
+      <div className="mx-20 flex flex-row justify-start items-center">
         <ScribbleIcon color={palette.turquoise[100]} />
         <Text type="h2">Available Subscriptions</Text>
       </div>
-      <div className="flex flex-col justify-center items-center py-12 gap-8">
+      <div className="flex flex-col justify-center items-center py-12 gap-8" data-theme="exampleUi">
```
Comment (juanmanso) [inline packages/nextjs/pages/index.tsx:72]: any reason why you are adding the `data-theme="exampleUi"` here?

File: packages/nextjs/pages/index.tsx [hunk 6]
```diff
@@ -3,6 +3,9 @@ import { SVGProps } from "react";
 import dynamic from "next/dynamic";
 import { useRouter } from "next/router";
 import { NextPage } from "next";
+import toast from "react-hot-toast";
```
Comment (juanmanso) [inline packages/nextjs/pages/index.tsx:6]: nitpick: when possible, use rebase instead of merge to retrieve changes from target branches. If you need I can show you how to do it, it's almost like merge but has its nuances

File: packages/nextjs/components/ProtectSection.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+import dynamic from "next/dynamic";
+import ProtectForm from "~~/components/ProtectForm";
+import { palette } from "~~/styles/colors";
+import { Text } from "~~/ui";
+
+const ScribbleIcon = dynamic<SVGProps<any>>(() => import("~~/public/assets/icons/scribble.svg"));
+
+const ProtectSection = () => {
+  return (
+    <div className="flex flex-col" style={{ backgroundColor: "transparent", padding: "0px 0px" }}>
```
Comment (juanmanso) [inline packages/nextjs/components/ProtectSection.tsx:11]: nitpick: no need for padding here now right?

File: packages/nextjs/ui/Text/Text.types.ts [hunk 8]
```diff
@@ -13,4 +13,4 @@ export type Typography =
   | "sm-print"
   | "btn-lg"
   | "btn-sm"
-  | "markdown";
+  | "markdown";
```
Comment (juanmanso) [inline packages/nextjs/ui/Text/Text.types.ts:16]: question: what are these break lines? Are they properly formatted?

same goes for `ui/index.ts`

File: packages/nextjs/components/ProtectSection.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+        <Text>{`Enter your site's URL to scan your robots.txt file`}</Text>
+        <div className="flex flex-row gap-2">
+          <ProtectForm />
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default ProtectSection;
```
Comment (juanmanso) [inline packages/nextjs/components/ProtectSection.tsx:26]: any reason to use default exports rather than following the convention? Is there any benefit?

File: packages/nextjs/components/ProtectSection.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import { SVGProps } from "react";
```
Comment (juanmanso) [inline packages/nextjs/components/ProtectSection.tsx:1]: Continuation of the comment on `ProtectForm`:

The `ProtectSection` is a part of the main landing page which is a very straight forward representational component with no much complexity. Moreover, this won't be used (or is not meant to be used) in other pages. So my question is, why are we doing the refactor for it? It even adds more code than it had before (~14 lines to 26 now)?

File: packages/nextjs/components/ProtectForm.tsx [hunk 11]
```diff
@@ -0,0 +1,60 @@
+import { FormEvent, useState } from "react";
```
Comment (juanmanso) [inline packages/nextjs/components/ProtectForm.tsx:1]: Comment about `ProtectSection` and `ProtectForm`:

These are confusing to me and let me tell you what goes on my mind when I'm facing them in the codebase as it does on the PR. On a quick glance I think "these are related to the protect flow", but are they?

Regarding the `ProtectForm` (`ProtectSection` is commented separately), does this refactor add any value? From what I'm understanding, it is just extracting it to a separated file just for the sake of it. 

That's good but if you want to reuse the component. The way that it is now, it's very scoped to just that use and does not improve the codebase other than separating the file and adding more lines to the codebase.

How about creating a `ui/Form` type of file where you implement the component there to be reused. Moreover, have you checked `ui/Input` to apply changes there if needed?

> _This is not to rain on your parade but just to be mindful of the changes and why we doing them. If there is a valid reason, please let me know and we can discuss them. Otherwise without any further explanation this seems unnecessary and entangles the codebase_

File: packages/nextjs/components/ProtectForm.tsx [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    }
+  };
+
+  return (
+    <form className="mt-8 text-black flex w-full" onSubmit={handleOnSubmit}>
+      <div className="flex-1 mr-2">
+        <Input
+          placeholder={INPUT_PLACEHOLDER}
+          value={url}
+          onChange={e => setUrl((e.target as HTMLInputElement).value)}
```
Comment (juanmanso) [inline packages/nextjs/components/ProtectForm.tsx:49]: what happened to the `keyDown`, `disabled` and other props here? any reason why they are not here? Is it because of the `form` component that wraps it, right? I'm not mega pro on that, so I'm asking because I don't know really 😅

File: packages/nextjs/pages/index.tsx [hunk 13]
```diff
@@ -149,13 +93,16 @@ const SubscriptionOverviewSection = () => {
...
                         </td>
                         <td className="border p-2 text-center">
-                          <Button color="secondary" onClick={() => browseToSubscriptionDetails(plan)}>
+                          <button
+                            className="btn btn-primary w-32 rounded-full capitalize font-normal font-white flex items-center transition-all tracking-widest"
+                            onClick={() => browseToSubscriptionDetails(plan)}
+                          >
                             More details
-                          </Button>
+                          </button>
```
Comment (juanmanso) [inline packages/nextjs/pages/index.tsx:105]: Is it worth it not creating a different `Button` style and implement a very specific button here instead? Isn't it very short scoped or am I missing something?

File: packages/nextjs/pages/subscribe/index.tsx [hunk 14]
```diff
@@ -50,7 +50,7 @@ const Subscribe = () => {
                         {plan.expirationBlock} Month{plan.expirationBlock !== "1" && "s"}
                       </td>
                       {/* <td className="border p-2">{plan.paymentTokenAddress}</td> */}
-                      <td className="border p-2 text-end">
+                      <td className="border p-2 text-center">
```
Comment (juanmanso) [inline packages/nextjs/pages/subscribe/index.tsx:53]: nitpick: would do this change on a different PR IMO but no need to remove it now, just a comment

File: packages/nextjs/ui/Button/Button.styles.ts [hunk 15]
```diff
@@ -21,7 +21,7 @@ export const colorButtonStyle: Record<ButtonColors, CSSProperties> = {
     backgroundColor: palette.purple[100],
   },
   secondary: {
-    backgroundColor: palette.slate[100],
+    backgroundColor: palette.purple[50],
```
Comment (juanmanso) [inline packages/nextjs/ui/Button/Button.styles.ts:24]: what happens with the background, won't do some weird stuff with the transparency?

**Review by juanmanso — COMMENTED** [review]
Added a few comments. Didn't approve because I would like to ask you for a couple more extra changes about the PR to follow the common formatting used here:
  a. Title following conventional commits
  b. Some description of the changes and/or the decisions made here
  c. Some sort of proof of the change. Either a screenshot if it's a static UI thing or a video if it's a functionality or behavioral change

File: packages/nextjs/components/ProtectSection.tsx (hunk 9 above)
Comment (huseyincansoylu) [reply packages/nextjs/components/ProtectSection.tsx:26]: no benefit , I think it's the extension that I use convert it, i will fix this

File: packages/nextjs/components/ProtectForm.tsx (hunk 11 above)
Comment (huseyincansoylu) [reply packages/nextjs/components/ProtectForm.tsx:1]: in 2 place we were using same functionality, calling same api , holding same state...
now we have only 1 component does the job , i think this refactoring adding value to the repo.

**Comment by huseyincansoylu** [comment]
> Added a few comments. Didn't approve because I would like to ask you for a couple more extra changes about the PR to follow the common formatting used here: a. Title following conventional commits b. Some description of the changes and/or the decisions made here c. Some sort of proof of the change. Either a screenshot if it's a static UI thing or a video if it's a functionality or behavioral change

issue has description 

I refactor the code , it is not ui thing
